src/main/java/de/tum/cit/aet/artemis/communication/service/conversation/ConversationDTOService.java

```diff
@@ -257,8 +258,15 @@ private GroupChatDTO convertGroupChatToDto(User requestingUser, GroupChat groupC
     @NotNull
     private Set<ConversationParticipant> getConversationParticipants(Conversation conversation) {
         Set<ConversationParticipant> conversationParticipants;
-        var participantsInitialized = Persistence.getPersistenceUtil().isLoaded(conversation, "conversationParticipants") && conversation.getConversationParticipants() != null
-                && !conversation.getConversationParticipants().isEmpty();
+        boolean participantsInitialized;
+        try {
+            participantsInitialized = Persistence.getPersistenceUtil().isLoaded(conversation, "conversationParticipants") && conversation.getConversationParticipants() != null
+                    && !conversation.getConversationParticipants().isEmpty();
+        }
+        catch (LazyInitializationException e) {
+            // In case the conversation's persistence context was closed already, we need to re-fetch to avoid errors down the line
+            participantsInitialized = false;
+        }
```

Comment:
After some digging I get why this is necessary, but I still find it a bit weird to need to have try/catch if before there is an explicit check to prevent exactly that exception.
Under which circumstances does this trigger? Maybe it might make sense to fix the issue there?

Reply:
I agree, both, the old and the new code do not look very good!

Reply:
We use `Hibernate.isInitialized(...)` in some places. Would that help to improve the code?

Hibernate.isInitialized(conversation.getConversationParticipants())